django/db/backends/mysql/base.py

```diff
@@ -270,6 +295,9 @@ def init_connection_state(self):
                 # for NULL. Disabling this brings this aspect of MySQL in line
                 # with SQL standards.
                 cursor.execute('SET SQL_AUTO_IS_NULL = 0')
+        if self.tx_isolation_level != 'repeatable read':  # change from MySQL default
+            with self.cursor() as cursor:
+                cursor.execute('SET SESSION TRANSACTION ISOLATION LEVEL %s' % self.tx_isolation_level)
```

Comment:
Ideally we wouldn't make an extra round trip on top of the `SQL_AUTO_IS_NULL` query above

MySQL also has the `init_command` option which sends queries on connection creation, we could glue this onto that, but that might be fragile.